logic/dapp.js

```diff
@@ -43,17 +43,19 @@ __private.unconfirmedAscii = {};
  * @param {Object} network
  * @todo Add description for the params
  */
-// Constructor
-function DApp(db, logger, schema, network) {
-	library = {
-		db,
-		logger,
-		schema,
-		network,
-	};
+class DApp {
+	constructor(db, logger, schema, network) {
+		library = {
+			db,
+			logger,
+			schema,
+			network,
+		};
+	}
 }
 
 // Public methods
+// TODO: TO maintain backward compatibility, have to user prototype otherwise these must be converted to static attributes
```

Comment:
So static class functions will not work out for backward compatibility?? 

```js
class DApp {
  constructor(db, logger, schema, network) {
    ...
  }
  static verify(transaction, sender, cb, tx) {
    ...
  }
}
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

Reply:
@tobiaslins in modules, tests and many other places we are accessing them through an instance of the class, changing it to static would lead to a lot of changes in the code the scope would go beyond `1.0.0` when we start doing modularization after `1.0.0` we can consider all the pain points we have in the system and use all possible latest es6 features.

Reply:
Ah okay, thanks for clarifying, but what about making them non-static class functions? Then they could be accessed in modules/tests! Or do I get something wrong?

Reply:
If we wanna make them non-static class functions, then we have to add eslint rule like following in each file
`/*eslint class-methods-use-this: ["error", { "exceptMethods": ["verify"] }] */`

Reply:
Just checked the implementation of these methods. Did not know that they aren't using this so it makes totally sense. I would go for your first recommendation and just use as many es6 features as possible without refactoring the coding (modularization and so on)
Thank you! :)